modules/series-service-impl/src/main/java/org/opencastproject/series/impl/persistence/SeriesServiceDatabaseImpl.java

```diff
@@ -383,6 +403,60 @@ public DublinCoreCatalog getSeries(String seriesId) throws NotFoundException, Se
     }
   }
 
+  @Override
+  public List<Series> getAllForAdministrativeRead(Date from, Optional<Date> to, int limit)
+          throws SeriesServiceDatabaseException, UnauthorizedException {
+    // Validate parameters
+    if (limit <= 0) {
+      throw new IllegalArgumentException("limit has to be > 0");
+    }
+
+    // Make sure the user is actually an administrator of sorts
+    User user = securityService.getUser();
+    if (!user.hasRole(GLOBAL_ADMIN_ROLE) && !user.hasRole(user.getOrganization().getAdminRole())) {
+      throw new UnauthorizedException(user, getClass().getName() + ".getModifiedInRangeForAdministrativeRead");
+    }
+
+    // Load series from DB.
+    EntityManager em = emf.createEntityManager();
+    try {
+      TypedQuery<SeriesEntity> q;
+      if (to.isPresent()) {
+        if (from.after(to.get())) {
+          throw new IllegalArgumentException("`from` is after `to`");
+        }
+
+        q = em.createNamedQuery("Series.getAllModifiedInRange", SeriesEntity.class)
+            .setParameter("from", from)
+            .setParameter("to", to.get())
+            .setMaxResults(limit);
+      } else {
+        q = em.createNamedQuery("Series.getAllModifiedSince", SeriesEntity.class)
+            .setParameter("since", from)
+            .setMaxResults(limit);
+      }
+
+      final List<Series> out = new ArrayList<>();
+      for (SeriesEntity entity : q.getResultList()) {
```

Comment:
`SeriesEntity` holds basically the same data the new `Series` does. Would it make sense to have `SeriesEntity` implement/extend `Series`?

Reply:
I don't really think so. `SeriesEntity` for me is just the connection to the DB and I would not want to mix it with `Series`. For example, `Series` contains `DublinCoreCatalog dublinCore`. If `SeriesEntity` would derive from `Series` that would have to be `null` all the time? And from experience, in these situations, sooner or later there will be more and more differences between the "database adapter" and the thing you want to use in your actual code.

Reply:
I would have probably solved that by making `Series` an interface and just requiring the methods, but fair enough, it's not really important.